app/components/move_message.py

```diff
@@ -124,28 +326,335 @@ async def on_submit(self, interaction: discord.Interaction) -> None:
+                reason=f"{interaction.user.name} wants to edit a moved message",
+                invitable=False,
+            )
+        except discord.Forbidden:
+            self.thread_button.disabled = True
+            await interaction.response.edit_message(content=NO_THREAD_PERMS, view=self)
+            raise  # Also log it in Sentry.
+        except discord.HTTPException as e:
+            if e.code != MAXIMUM_NUMBER_OF_ACTIVE_THREADS_REACHED:
+                # None of the other errors are relevant here.
+                raise
+            self.thread_button.disabled = True
+            await interaction.response.edit_message(content=TOO_MANY_THREADS, view=self)
+            return
+        # Notify the user as soon as possible to prevent the need to defer() to
+        # avoid issues on days when the API is slower.
+        await interaction.response.edit_message(
+            content=f"Created a thread: {thread.mention}.", view=None
+        )
+        if self._split_subtext.content:
+            await thread.send(
+                f"{interaction.user.mention}, here are the contents of your message:"
+            )
+            await thread.send(
+                self._split_subtext.content,
+                allowed_mentions=discord.AllowedMentions.none(),
+            )
+            await thread.send(EDIT_IN_THREAD_HINT, view=CancelEditing(thread))
+        else:
+            await thread.send(
+                NO_CONTENT_TO_EDIT.format(interaction.user.mention),
+                view=CancelEditing(thread),
+            )
+        edit_threads[thread.id] = ThreadState(
+            self._message, self._split_subtext, dt.datetime.now(tz=dt.UTC)
+        )
+        edit_thread_creators[interaction.user.id] = thread.id
+        await _remove_edit_thread_after_timeout(thread, interaction.user)
 
-    @discord.ui.button(
-        label="Delete instead",
-        emoji="🗑️",  # test: allow-vs16
-        style=discord.ButtonStyle.danger,
+    async def show_help(self, interaction: discord.Interaction) -> None:
+        self.help_button.disabled = True
+        await interaction.response.edit_message(
+            content=f"{MESSAGE_EDIT_HELP}\n\n{EDIT_METHOD_PROMPT}", view=self
+        )
+
+
+class EditMessage(discord.ui.Modal, title="Edit Message"):
+    new_text: discord.ui.TextInput[Self] = discord.ui.TextInput(
+        label="New message content",
+        style=discord.TextStyle.long,
+        default=discord.utils.MISSING,
     )
-    async def delete(
-        self,
-        interaction: discord.Interaction,
-        button: discord.ui.Button[Self],
+
+    def __init__(self, message: MovedMessage, split_subtext: SplitSubtext) -> None:
+        self._split_subtext = split_subtext
+        self.new_text.default = split_subtext.content
+        # Subtract one to account for the newline character.
+        self.new_text.max_length = 2000 - len(split_subtext.subtext) - 1
+        super().__init__()
```

Comment:
Any reason this `super` call is not the first thing done in the `__init__`? Feels off to me.